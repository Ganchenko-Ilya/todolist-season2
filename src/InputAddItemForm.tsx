import { ChangeEvent, KeyboardEvent, useState } from "react";
import s from "./InputAddTask.module.css";

import AddIcon from "@mui/icons-material/Add";
import { TextField } from "@mui/material";
type InputAddTaskProps = {
  addItem: (value: string) => void;
  placeholder?: string;
};
export const InputAddItemForm = (props: InputAddTaskProps) => {
  const { addItem, placeholder } = props;
  const [value, setValue] = useState<string>("");
  const [error, setError] = useState<string>("");
  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.currentTarget.value);
    setError("");
  };
  const onClickOnKeyDownHandler = () => {
    if (value.trim()) {
      addItem(value);
    } else {
      setError("Incorrect input");
    }
    setValue("");
  };
  const onClickHandler = () => {
    onClickOnKeyDownHandler();
  };
  const onKeyDownHandler = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter") onClickOnKeyDownHandler();
  };
  return (
    <div className={s.wrapper}>
      <TextField
        value={value}
        placeholder={placeholder}
        error={!!error}
        variant="outlined"
        onKeyDown={onKeyDownHandler}
        onChange={onChangeHandler}
      />
      <AddIcon onClick={onClickHandler} />
      <div className={s.textError}>{error}</div>
    </div>
  );
};
