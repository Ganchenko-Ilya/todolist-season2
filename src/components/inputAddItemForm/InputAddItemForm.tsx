import { ChangeEvent, KeyboardEvent, useState } from "react";
import s from "./InputAddItemForm.module.css";

import AddIcon from "@mui/icons-material/Add";
import { TextField } from "@mui/material";
type InputAddTaskProps = {
  addItem: (value: string) => void;
  helpText:string
};
export const InputAddItemForm = (props: InputAddTaskProps) => {
  const { addItem, helpText } = props;
  const [value, setValue] = useState<string>("");
  const [error, setError] = useState<string>("");
  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.currentTarget.value);
    setError("");
  };
  const onClickOnKeyDownHandler = () => {
    if (value.trim()) {
      addItem(value);
    } else {
      setError("Incorrect input");
    }
    setValue("");
  };
  const onClickHandler = () => {
    onClickOnKeyDownHandler();
  };
  const onKeyDownHandler = (e: KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter") onClickOnKeyDownHandler();
  };
  return (
    <div className={s.wrapper}>
      <TextField
        className={error ? s.inputError : ''}
        value={value}
        placeholder={error ? error : helpText}
        error={!!error}
        variant="outlined"
        onKeyDown={onKeyDownHandler}
        onChange={onChangeHandler}
      />
      <AddIcon onClick={onClickHandler} />
    </div>
  );
};
