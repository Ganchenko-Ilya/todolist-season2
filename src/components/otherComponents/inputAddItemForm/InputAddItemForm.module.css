.wrapper {
  display: flex;
  align-items: center;
}
.wrapper .inputError input::placeholder {
  color: red;
}
