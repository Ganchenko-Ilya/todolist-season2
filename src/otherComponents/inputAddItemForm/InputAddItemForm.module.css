.wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
}
.wrapper .inputError input::placeholder {
  color: red;
}
