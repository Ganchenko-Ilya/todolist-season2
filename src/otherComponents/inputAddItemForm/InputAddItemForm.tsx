import React, { ChangeEvent, KeyboardEvent, useCallback, useState } from "react";
import s from "./InputAddItemForm.module.css";

import AddIcon from "@mui/icons-material/Add";
import { TextField } from "@mui/material";
import { StatusType } from "../../store/app-reducer";
type InputAddTaskProps = {
  addItem: (value: string) => void;
  helpText: string;
  statusTodo?: StatusType;
};
export const InputAddItemForm = React.memo((props: InputAddTaskProps) => {
  console.log("InputAddItemForm");
  const { addItem, helpText, statusTodo } = props;
  const [value, setValue] = useState<string>("");
  const [error, setError] = useState<string>("");
  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.currentTarget.value);
    if (error) setError("");
  };
  const onClickOnKeyDownHandler = () => {
    if (value.trim()) {
      addItem(value);
    } else {
      setError("Incorrect input");
    }
    setValue("");
  };
  const onClickHandler = useCallback(() => {
    onClickOnKeyDownHandler();
  }, [onClickOnKeyDownHandler]);
  const onKeyDownHandler = useCallback(
    (e: KeyboardEvent<HTMLInputElement>) => {
      if (e.key === "Enter") onClickOnKeyDownHandler();
    },
    [onClickOnKeyDownHandler]
  );

  return (
    <div className={s.wrapper}>
      <TextField
        className={error ? s.inputError : ""}
        value={value}
        placeholder={error ? error : helpText}
        error={!!error}
        variant="outlined"
        onKeyDown={onKeyDownHandler}
        onChange={onChangeHandler}
        disabled={statusTodo === "loading"}
      />
      <AddIcon
        onClick={onClickHandler}
        style={
          statusTodo === "loading"
            ? { pointerEvents: "none", cursor: "pointer" }
            : { cursor: "pointer" }
        }
      />
    </div>
  );
});
