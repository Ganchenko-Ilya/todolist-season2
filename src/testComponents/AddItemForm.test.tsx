import renderer from "react-test-renderer";
import { InputAddItemForm } from "../components/otherComponents/inputAddItemForm/InputAddItemForm";




test("test code Components", () => {
  const result = renderer.create(<InputAddItemForm addItem={() => {}} helpText="add" />).toJSON();
  expect(result).toMatchSnapshot();
});