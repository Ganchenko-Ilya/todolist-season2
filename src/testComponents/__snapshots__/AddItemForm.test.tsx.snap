// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test code Components 1`] = `
<div
  className="wrapper"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    onKeyDown={[Function]}
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="add"
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="AddIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
    />
  </svg>
</div>
`;
